invenio_rdm_records/stats/event_builders.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio RDM Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Custom event builders for the InvenioRDM statistics.
+
+Note that the arguments to these functions will be the same as passed to the
+``EventEmitter`` objects when they are called.
+Currently in InvenioRDM this is done in resources (for API) and view functions (for UI).
+As such, it is assumed that a request context is available (and thus, Flask's global
+``request`` is accessible).
+"""
+
+from datetime import datetime
+
+from flask import current_app, request
+from invenio_stats.utils import get_user
+
+
+def file_download_event_builder(event, sender_app, **kwargs):
+    """Build a file-download event.
+
+    *Note* that this function assumes a request context by accessing properties of
+    Flask's global ``request`` object.
+    """
+    assert "record" in kwargs
+    assert "obj" in kwargs
+
+    record = kwargs["record"]
+    obj = kwargs["obj"]
+    event.update(
+        {
+            # When:
+            "timestamp": datetime.utcnow().isoformat(),
+            # What:
+            "bucket_id": str(obj.bucket_id),
+            "file_id": str(obj.file_id),
+            "file_key": obj.key,
+            "size": obj.file.size,
+            "recid": record.get("id", None),
```

Comment:
could there be a case of both recird and parent being "None"? Or even just one of them? Even if it is a file related event there would be an associated record? I'm just wondering if we would check if one or both exists actually... It affect other event builders in this file.

Reply:
tbh, i don't think that either of them being `None` could ever be the case during normal operation...
if it were, chances are that accessing `record.parent` would also fail.
should we change the accessor to `record["id"]` and `record.parent["id"]`?